packages/mds-audit-api/api.ts

```diff
@@ -676,27 +676,42 @@ function api(app: express.Express): express.Express {
     pathPrefix('/vehicles'),
     checkAuditApiAccess(scopes => scopes.includes('audits:vehicles:read')),
     async (req: AuditApiGetVehicleRequest, res: GetAuditVehiclesResponse) => {
-      const { skip, take } = { skip: 0, take: 10000 }
-      const {
-        strict = true,
-        bbox,
-        provider_id
-      } = {
-        ...parseRequest(req).single({ parser: JSON.parse }).query('strict', 'bbox'),
-        ...parseRequest(req).single().query('provider_id')
-      }
+      try {
+        const { skip, take } = { skip: 0, take: 10000 }
+        const {
+          strict = true,
+          bbox,
+          provider_id
+        } = {
+          ...parseRequest(req)
+            .single({
+              parser: (val, key) => {
+                try {
+                  return JSON.parse(val)
+                } catch (error) {
+                  throw new ValidationError(`invalid query parameter ${key}`, { [key]: val })
```

Comment:
In general it's a bad idea to return any user input back to them, this is probably fine, but watch out, this is the basis for most buffer or injection attacks. Less helpful error messages are the only way.

Reply:
Can we log the error with the key/val here, but change the error message in the exception thrown to something more cryptic perhaps?

Reply:
Considering the clients for this API (and basically all others we operate right now) are at least semi-trusted parties that've auth'd, I'm a bit less worried about that scenario. Generally agreed, though.

Reply:
At the end of the day, I think `key` can only be from a list provided in our code here, and so could not be 'evil'.